src/components/leader-table.tsx

```diff
@@ -0,0 +1,395 @@
+    createData('https://via.placeholder.com/64/0000FF/808080?Text=Examle', 'Name', 'Nickname', 140),
+    createData('https://via.placeholder.com/64/0000FF/808080?Text=Examle', 'Name', 'Nickname', 155),
+    createData('https://via.placeholder.com/64/0000FF/808080?Text=Examle', 'Name', 'Nickname', 121),
+    createData('https://via.placeholder.com/64/0000FF/808080?Text=Examle', 'Name', 'Nickname', 234),
+    createData('https://via.placeholder.com/64/0000FF/808080?Text=Examle', 'Name', 'Nickname', 854),
+    createData('https://via.placeholder.com/64/0000FF/808080?Text=Examle', 'Name', 'Nickname', 709),
+    createData('https://via.placeholder.com/64/0000FF/808080?Text=Examle', 'Name', 'Nickname', 231),
+];
+
+function descendingComparator<T>(a: T, b: T, orderBy: keyof T) {
+    if (b[orderBy] < a[orderBy]) {
+        return -1;
+    }
+    if (b[orderBy] > a[orderBy]) {
+        return 1;
+    }
+    return 0;
+}
+
+type Order = 'asc' | 'desc';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function getComparator<Key extends keyof any>(
+    order: Order,
+    orderBy: Key,
+): (
+    a: { [key in Key]: number | string },
+    b: { [key in Key]: number | string },
+) => number {
+    return order === 'desc'
+        ? (a, b) => descendingComparator(a, b, orderBy)
+        : (a, b) => -descendingComparator(a, b, orderBy);
+}
+
+// This method is created for cross-browser compatibility, if you don't
+// need to support IE11, you can use Array.prototype.sort() directly
+function stableSort<T>(array: ReadonlyArray<T>, comparator: (a: T, b: T) => number) {
+    const stabilizedThis = array.map((el, index) => [el, index] as [T, number]);
+    stabilizedThis.sort((a, b) => {
+        const order = comparator(a[0], b[0]);
+        if (order !== 0) {
+            return order;
+        }
+        return a[1] - b[1];
+    });
+    return stabilizedThis.map(el => el[0]);
+}
+
+interface HeadCell {
+    disablePadding: boolean;
+    id: keyof Data;
+    label: string;
+    numeric: boolean;
+}
+
+const headCells: ReadonlyArray<HeadCell> = [
+    {
+        id: 'avatar',
+        numeric: false,
+        disablePadding: true,
+        label: 'Аватар',
+    },
+    {
+        id: 'name',
+        numeric: false,
+        disablePadding: false,
+        label: 'Имя',
+    },
+    {
+        id: 'nickname',
+        numeric: false,
+        disablePadding: false,
+        label: 'Никнейм',
+    },
+    {
+        id: 'record',
+        numeric: true,
+        disablePadding: false,
+        label: 'Рекорд',
+    },
+];
+
+interface EnhancedTableProps {
+    numSelected: number;
+    onRequestSort: (event: React.MouseEvent<unknown>, property: keyof Data) => void;
+    onSelectAllClick: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    order: Order;
+    orderBy: string;
+    rowCount: number;
+}
+
+function EnhancedTableHead(props: EnhancedTableProps) {
```

Comment:
nit: Деструктуризацию можно делать прямо в объявлении компонента

```suggestion
function EnhancedTableHead({onSelectAllClick, order, orderBy, numSelected, rowCount, onRequestSort}: EnhancedTableProps) {
```